invenio_rdm_migrator/load/postgresql.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""PostgreSQL load module."""
+
+
+import contextlib
+import json
+import uuid
+from abc import ABC, abstractmethod
+from dataclasses import fields
+from datetime import datetime
+from pathlib import Path
+
+import psycopg
+from .base import Load
+
+
+def _ts(iso=True):
+    """Current timestamp string."""
+    dt = datetime.now()
+    return dt.isoformat() if iso else dt.timestamp()
+
+
+def as_csv_row(dc):
+    """Serialize a dataclass instance as a CSV-writable row."""
+    row = []
+    for f in fields(dc):
+        val = getattr(dc, f.name)
+        if val:
+            if issubclass(f.type, (dict,)):
+                val = json.dumps(val)
+            elif issubclass(f.type, (datetime,)):
+                val = val.isoformat()
+            elif issubclass(f.type, (uuid.UUID,)):
+                val = str(val)
+        row.append(val)
+    return row
+
+
+class PostgreSQLCopyLoad(Load):  # TODO: abstract SQL from PostgreSQL?
+    """PostgreSQL COPY load."""
+
+    def __init__(self, db_uri, table_loads, output_path):
+        """Constructor."""
+        self.db_uri = db_uri
+        self.output_dir = Path(output_path) / f"data/tables{_ts(iso=False)}"
+        self.output_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
minor: not a big fan of "doing things" in the constructor. maybe we can do that right before the folder is used?

Reply:
makes sense, moved the _mkdir_ to the `_prepare` method which is where we would dump files.